PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/intellij/runner/webapp/webappconfig/slimui/creation/WebAppCreationDialog.java

```diff
@@ -0,0 +1,434 @@
+    }
+
+    private void createWebApp() {
+        updateConfiguration();
+        ProgressManager.getInstance().run(new Task.Modal(null, "Creating New WebApp...", true) {
+            @Override
+            public void run(ProgressIndicator progressIndicator) {
+                try {
+                    progressIndicator.setIndeterminate(true);
+                    result = AzureWebAppMvpModel.getInstance().createWebApp(webAppConfiguration.getModel());
+                    dispose();
+                } catch (Exception ex) {
+                    JOptionPane.showMessageDialog(null, "Create WebApp Failed : " + ex.getMessage(), "Create WebApp " +
+                            "Failed",
+                        JOptionPane.ERROR_MESSAGE);
+                }
+            }
+        });
+    }
+
+    @Override
+    public void fillSubscription(@NotNull List<Subscription> subscriptions) {
+        fillCombobox(this, cbSubscription, subscriptions, null);
+    }
+
+    @Override
+    public void fillResourceGroup(@NotNull List<ResourceGroup> resourceGroups) {
+        fillCombobox(this, cbExistResGrp, resourceGroups, null);
+    }
+
+    @Override
+    public void fillAppServicePlan(@NotNull List<AppServicePlan> appServicePlans) {
+        cbExistAppServicePlan.removeAllItems();
+        appServicePlans.stream()
+            .filter(item -> Comparing.equal(item.operatingSystem(), webAppConfiguration.getOS()))
+            .sorted(Comparator.comparing(AppServicePlan::name))
+            .forEach((plan) -> {
+                cbExistAppServicePlan.addItem(plan);
+            });
+        selectAppServicePlan();
+        pack();
+    }
+
+    @Override
+    public void fillLocation(@NotNull List<Location> locations) {
+        cbLocation.removeAllItems();
+        locations.stream()
+            .sorted(Comparator.comparing(Location::displayName))
+            .forEach((location) -> {
+                cbLocation.addItem(location);
+                if (Comparing.equal(location.name(), DEFAULT_REGION.name())) {
+                    cbLocation.setSelectedItem(location);
+                }
+            });
+        pack();
+    }
+
+    @Override
+    public void fillPricingTier(@NotNull List<PricingTier> prices) {
+        fillCombobox(this, cbPricing, prices, DEFAULT_PRICINGTIER);
+    }
+
+    @Override
+    public void fillWebContainer(@NotNull List<WebAppUtils.WebContainerMod> webContainers) {
+        fillCombobox(this, cbWebContainer, webContainers, DEFAULT_WINDOWS_CONTAINER);
+    }
+
+    @Override
+    public void fillJdkVersion(@NotNull List<JdkModel> jdks) {
+        fillCombobox(this, cbJdkVersion, jdks, DEFAULT_WINDOWS_JAVAVERSION);
+    }
+
+    @Override
+    public void fillLinuxRuntime(@NotNull List<RuntimeStack> linuxRuntimes) {
+        fillCombobox(this, cbRuntimeStack, linuxRuntimes, DEFAULT_LINUX_RUNTIME);
+    }
+
+    public static <T> void fillCombobox(Window window, JComboBox<T> comboBox, List<T> values, T defaultValue) {
```

Comment:
why is it public for external package?

Reply:
Plan to make it as an common method before, will fix it as other fill operation are more complex.